server/render/index.js

```diff
@@ -40,25 +41,64 @@ function bumpStat( group, name ) {
 	}
 }
 
+export function getCacheKey( context ) {
+	const pathname = context.pathname;
+
+	if ( isEmpty( context.query ) || isEmpty( context.cacheQueryKeys ) ) {
+		return pathname;
+	}
+
+	const cachedQueryParams = pick( context.query, context.cacheQueryKeys );
+
+	if ( isEmpty( cachedQueryParams ) ) {
+		return pathname;
+	}
+
+	return pathname + '?' + qs.stringify( cachedQueryParams, { sort: ( a, b ) => a.localCompare( b ) } );
```

Comment:
is `localCompare` supposed to be `localeCompare`?

Reply:
Quite likely :slightly_smiling_face:

Reply:
Good catch! I guess we never actually tested that code since we never had more than 1 cachedQueryParams so far :s